lib/datapacksjob.js

```diff
@@ -3190,4 +3193,57 @@ DataPacksJob.prototype.getOrgProjects = async function (jobInfo) {
         jobInfo.hasError = true;
         jobInfo.errors.push(err);
     }
+}
+
+DataPacksJob.prototype.getUrlByActionDetails = async function (trackingEntries) {
+    let actionsConfig = this.vlocity.datapacksutils.dataPacksExpandedDefinition.IPActionsConfig;
+    let tempTrackingEntries = [];
+    let isItemString = typeof (trackingEntries[0]) === 'string';
+    for (let entry of trackingEntries) {
+        entry = isItemString ? JSON.parse(entry) : entry;
+        if (entry.TrackingEvent && entry.TrackingEvent === "StepActionTime") {
+            if (actionsConfig[entry.ElementType]) {
+                let config = actionsConfig[entry.ElementType];
+                let query = config.query.replace(/%vlocity_namespace%__/g, this.vlocity.namespacePrefix);
+                let queryAddOns = '';
+                if (config.nameField === 'integrationProcedureKey') {
```

Comment:
There is actually a ProcedureKey__c field on the OmniScript__c. Probably best to add a Where field to the configuration data.